portable/GCC/MicroBlazeV9/portasm.S

```diff
@@ -156,6 +178,15 @@ back into the caller stack. */
 	lwi r18, r0, pxCurrentTCB
 	lw	r1, r0, r18
 
+#if( XPAR_MICROBLAZE_USE_STACK_PROTECTION )
+	/* Restore the stack limits -- must not load from r1! */

```

Comment:
```suggestion
	/* Restore the stack limits -- must not load from r1 (Stack Pointer)
	 * because if the address of load or store instruction is out of range,
	 * it will trigger  Stack Protection Violation exception.  */

```